dev-packages/application-manager/package.json

```diff
@@ -35,25 +35,24 @@
     "@theia/application-package": "1.13.0",
     "@theia/compression-webpack-plugin": "^3.0.0",
     "@types/fs-extra": "^4.0.2",
-    "@types/webpack": "^4.41.2",
-    "babel-loader": "^8.0.6",
-    "circular-dependency-plugin": "^5.0.0",
-    "copy-webpack-plugin": "^4.5.0",
+    "babel-loader": "^8.2.2",
+    "buffer": "^6.0.3",
+    "circular-dependency-plugin": "^5.2.2",
+    "copy-webpack-plugin": "^8.1.1",
     "css-loader": "^0.28.1",
     "electron-rebuild": "^1.8.6",
-    "file-loader": "^1.1.11",
     "font-awesome-webpack": "0.0.5-beta.2",
     "fs-extra": "^4.0.2",
     "ignore-loader": "^0.1.2",
     "less": "^3.0.3",
-    "source-map-loader": "^0.2.1",
-    "source-map-support": "^0.4.18",
-    "style-loader": "^0.23.1",
-    "umd-compat-loader": "^2.1.1",
-    "url-loader": "^1.1.2",
-    "webpack": "^4.0.0",
-    "webpack-cli": "2.0.12",
-    "worker-loader": "^1.1.1",
+    "setimmediate": "^1.0.5",
+    "source-map-loader": "^2.0.1",
+    "source-map-support": "^0.5.19",
+    "style-loader": "^2.0.0",
```

Comment:
@paul-marechal One thing I still don't understand here is why these are not in the `devDependencies` section. Does webpack not take care of including them in the mangled output?

Reply:
`devDependencies` are not installed when a package is depended upon meaning those packages would be missing for dependents. Hence those being regular dependencies.

Regarding the bundling, `setimmediate` is required in `src-gen` and bundled. The loaders should never reach the final bundles, although they need to be in the environment used to run Webpack (using Theia CLI.)

Reply:
Thinking about it, if `@theia/application-manager` only appears as a `dev-dependency` in a application `package.json` it should keep working indeed. But from the POV of `@theia/application-manager` these dependencies must be pulled alongside it so they can't be declared as `dev-dependencies`.

Reply:
Ok, so the correct way to consume all that is to have `@theia/application-manager` as a dev dependency in the application package (like `examples/browser`)?

Reply:
Experimenting a bit further with this: unless I add an explicit dependency in my application to stuff like `circular-dependency-plugin` only gets installed into `node_modules/@theia/application-manager/node_modules/circular-dependency-plugin` and thus is not available for `require(...)`from `gen-weback-config.js`. So I think we need to fix https://theia-ide.org/docs/composing_applications.

Reply:
What you describe sounds like a hoisting problem and it can be painful to troubleshoot. It is part of why I wanted to limit implicit dependencies as much as possible. In this specific case I am not sure how you could dedup the package. `yarn why` can be useful to see all dependents, and if every package depends on the same version it should install it only once at the root of your `node_modules`.

Reply:
How do you folks build your application packages? Do you consume the theia packages as npm modules?

Reply:
Most of the time yes.